### PR TITLE
Con 2206 tracking subscriptions

### PR DESCRIPTION
**Pull request by Carsten89**

### CODE REVIEW

File: product/selectors/page.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
+/**
```
Comment (richardgorman) [inline product/selectors/page.js]: Please add periods where appropriate throughout the file.

File: product/selectors/product.js [hunk 2]
```diff
@@ -10,18 +10,13 @@ import { generateResultHash } from '@shopgate/pwa-common/helpers/redux';
...
 import { getActiveFilters } from '../../filter/selectors';
 import { getCurrentCategoryId } from '../../category/selectors';
-import {
-  hasCurrentProductVariants,
-  getCurrentBaseProductVariants,
-  getCurrentProductVariantId,
-} from './variants';
 
 /**
  * Selects collection of all stored products from the store.
```
Comment (richardgorman) [inline product/selectors/product.js]: `Selects all products from the store.`

File: product/streams/index.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  .filter(({ action }) => action.type === RECEIVE_PRODUCT)
+  .distinctUntilChanged();
+
+/**
+ * Gets triggered when VariantId changes
+ * @type {Observable}
+ */
+const setVariantId$ = main$
+  .filter(({ action }) => (
+    action.type === SET_PRODUCT_VARIANT_ID && action.productVariantId !== null
```
Comment (richardgorman) [inline product/streams/index.js]: Please make this multiline.

File: product/streams/index.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+/**
+ * Gets triggered when product data received.
+ * @type {Observable}
+ */
+export const productReceived$ = main$
+  .filter(({ action }) => action.type === RECEIVE_PRODUCT)
+  .distinctUntilChanged();
+
+/**
+ * Gets triggered when VariantId changes
```
Comment (richardgorman) [inline product/streams/index.js]: Missing period.

File: product/streams/index.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+ */
+export const variantDidChangeUncached$ = setVariantId$
+  .switchMap(({ action }) => (
+    productReceived$.filter(
+      ({ action: productAction }) => productAction.productId === action.productVariantId)
+    )
+  );
+
+/**
+ * Gets triggered when VariantId changes and product data are already there.
```
Comment (richardgorman) [inline product/streams/index.js]: ... is already there.

File: product/streams/index.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+ * @type {Observable}
+ */
+const variantDidChangedCached = setVariantId$
+  .filter(({ getState }) => !!getSelectedVariant(getState()));
+
+/**
+ * Gets triggered when VariantId changes and product data are available.
```
Comment (richardgorman) [inline product/streams/index.js]: ... is already there.

File: product/streams/index.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+import 'rxjs/add/operator/switchMap';
+import { main$ } from '@shopgate/pwa-common/streams/main';
+import {
+  RECEIVE_PRODUCT,
+  SET_PRODUCT_VARIANT_ID,
+} from '../constants';
+import { getSelectedVariant } from '../selectors/variants';
+
+/**
+ * Gets triggered when product data received.
```
Comment (richardgorman) [inline product/streams/index.js]: ... data is received.

File: product/selectors/variants.js [hunk 8]
```diff
@@ -127,3 +102,26 @@ export const getProductVariants = createSelector(
     return variants.variants;
   }
 );
+
+/**
+ * Retrieves the current selected product variant.
+ * @param {Object} state The application state.
+ * @returns {Object|null} The selected variant or null if non is selected
```
Comment (richardgorman) [inline product/selectors/variants.js]: Type: non

File: product/selectors/page.js (hunk 1 above)
Comment (Carsten89) [reply product/selectors/page.js]: periods: OK
re comment all the stuff: out of scope. i just moved it

File: product/selectors/product.js (hunk 2 above)
Comment (Carsten89) [reply product/selectors/product.js]: done

File: product/streams/index.js (hunk 3 above)
Comment (Carsten89) [reply product/streams/index.js]: done

File: product/streams/index.js (hunk 4 above)
Comment (Carsten89) [reply product/streams/index.js]: done

File: product/streams/index.js (hunk 5 above)
Comment (Carsten89) [reply product/streams/index.js]: done

File: product/streams/index.js (hunk 6 above)
Comment (Carsten89) [reply product/streams/index.js]: done

File: product/streams/index.js (hunk 7 above)
Comment (Carsten89) [reply product/streams/index.js]: done

File: product/selectors/variants.js (hunk 8 above)
Comment (Carsten89) [reply product/selectors/variants.js]: done

**Comment by richardgorman** [comment]
When I navigate to the root category page from the index, the page is tracked twice.

File: product/selectors/product.js [hunk 9]
```diff
@@ -10,18 +10,13 @@ import { generateResultHash } from '@shopgate/pwa-common/helpers/redux';
...
 import { getCurrentCategoryId } from '../../category/selectors';
-import {
-  hasCurrentProductVariants,
-  getCurrentBaseProductVariants,
-  getCurrentProductVariantId,
-} from './variants';
 
 /**
- * Selects collection of all stored products from the store.
+ * SSelects all products from the store.
```
Comment (richardgorman) [inline product/selectors/product.js]: Typo: `SS`

File: product/selectors/page.js (hunk 1 above)
Comment (richardgorman) [reply product/selectors/page.js]: That may be true, but the comments suck and we are already here. To me it makes sense to rewrite these comments now.

File: product/selectors/product.js (hunk 9 above)
Comment (Carsten89) [reply product/selectors/product.js]: done

**Pull request merged**